moveit_ros/moveit_servo/src/cpp_interface_demo/pose_tracking_demo.cpp

```diff
@@ -78,82 +82,107 @@ class StatusMonitor
+
+  rclcpp::sleep_for(std::chrono::seconds(5));
 
   // Create the pose tracker
-  moveit_servo::PoseTracking tracker(nh, planning_scene_monitor);
+  moveit_servo::PoseTracking tracker(node, parameters, planning_scene_monitor);
 
   // Make a publisher for sending pose commands
-  ros::Publisher target_pose_pub =
-      nh.advertise<geometry_msgs::PoseStamped>("target_pose", 1 /* queue */, true /* latch */);
+  rclcpp::Publisher<geometry_msgs::msg::PoseStamped>::SharedPtr target_pose_pub;
+  target_pose_pub =
+      node->create_publisher<geometry_msgs::msg::PoseStamped>("target_pose", 1 /* queue */);
 
   // Subscribe to servo status (and log it when it changes)
-  StatusMonitor status_monitor(nh, "status");
+  StatusMonitor status_monitor(node, parameters->status_topic);
 
-  Eigen::Vector3d lin_tol{ 0.01, 0.01, 0.01 };
-  double rot_tol = 0.1;
+  Eigen::Vector3d lin_tol{ 0.001, 0.001, 0.001 };
+  double rot_tol = 0.01;
 
   // Get the current EE transform
-  geometry_msgs::TransformStamped current_ee_tf;
+  geometry_msgs::msg::TransformStamped current_ee_tf;
   tracker.getCommandFrameTransform(current_ee_tf);
 
   // Convert it to a Pose
-  geometry_msgs::PoseStamped target_pose;
+  geometry_msgs::msg::PoseStamped target_pose;
   target_pose.header.frame_id = current_ee_tf.header.frame_id;
   target_pose.pose.position.x = current_ee_tf.transform.translation.x;
   target_pose.pose.position.y = current_ee_tf.transform.translation.y;
   target_pose.pose.position.z = current_ee_tf.transform.translation.z;
   target_pose.pose.orientation = current_ee_tf.transform.rotation;
 
+
   // Modify it a little bit
-  target_pose.pose.position.x += 0.1;
+  target_pose.pose.position.x += 0.01;
 
   // resetTargetPose() can be used to clear the target pose and wait for a new one, e.g. when moving between multiple
   // waypoints
   tracker.resetTargetPose();
 
   // Publish target pose
-  target_pose.header.stamp = ros::Time::now();
-  target_pose_pub.publish(target_pose);
+  target_pose.header.stamp = node->now();
+  target_pose_pub->publish(target_pose);
 
   // Run the pose tracking in a new thread
   std::thread move_to_pose_thread(
-      [&tracker, &lin_tol, &rot_tol] { tracker.moveToPose(lin_tol, rot_tol, 0.1 /* target pose timeout */); });
+      [&tracker, &lin_tol, &rot_tol] { 
+      moveit_servo::PoseTrackingStatusCode tracking_status = tracker.moveToPose(lin_tol, rot_tol, 0.1 /* target pose timeout */); 
+      RCLCPP_INFO_STREAM(LOGGER, "Pose tracker exited with status: " << moveit_servo::POSE_TRACKING_STATUS_CODE_MAP.at(tracking_status));
+      });
 
-  ros::Rate loop_rate(50);
+  rclcpp::Rate loop_rate(50);
   for (size_t i = 0; i < 500; ++i)
   {
     // Modify the pose target a little bit each cycle
     // This is a dynamic pose target
-    target_pose.pose.position.z += 0.0004;
-    target_pose.header.stamp = ros::Time::now();
-    target_pose_pub.publish(target_pose);
+    target_pose.pose.position.z += 0.0001;
```

Comment:
That makes my arm overshoot and so there is an emergency stop (since it gets very close to a singularity).
I thought the position is in terms of meters but I'm not sure looking at how far the arm reaches out with just 0.001 change in the z direction on RViz.